### PR TITLE
INTYGFV-15113: Refactored ids for AbstractQuestionDiagnoser to match field in validationErrors.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
Also fixed some wrongly used questionIds.

### CODE REVIEW

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java [hunk 1]
```diff
@@ -147,8 +150,43 @@ protected static List<Diagnos> toInternal(Certificate certificate, String questi
...
+    private static int convertIdToInt(String id) {
+        switch (id) {
+            case DIAGNOSES_LIST_ITEM_1_ID:
+                return 0;
+            case DIAGNOSES_LIST_ITEM_2_ID:
+                return 1;
+            case DIAGNOSES_LIST_ITEM_3_ID:
+                return 2;
+        }
+        return 0;
```
Comment (mikeyOne) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java:178]: Could use a switch-default instead.

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java [hunk 2]
```diff
@@ -147,8 +150,43 @@ protected static List<Diagnos> toInternal(Certificate certificate, String questi
...
+    private static String convertIntToId(String id) {
+        switch (id) {
+            case "1":
+                return DIAGNOSES_LIST_ITEM_1_ID;
+            case "2":
+                return DIAGNOSES_LIST_ITEM_2_ID;
+            case "3":
+                return DIAGNOSES_LIST_ITEM_3_ID;
+        }
+        return "";
```
Comment (mikeyOne) [inline fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java:190]: Could use a switch-default instead.

**Review by mikeyOne — APPROVED** [review]

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Good suggestion! Done!

File: fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply fk/fk-parent/src/main/java/se/inera/intyg/common/fkparent/model/converter/certificate/AbstractQuestionDiagnoser.java]: Good suggestion! Done!

**Pull request merged**